code/modules/ninja/suit/ninjaDrainAct.dm

```diff
@@ -144,14 +152,20 @@ They *could* go in their appropriate files, but this is supposed to be modular
 
 	. = DRAIN_RD_HACK_FAILED
 
-	to_chat(H, span_notice("Hacking \the [src]..."))
+	to_chat(H, span_notice("Research notes detected. Corrupting data..."))
 	AI_notify_hack()
 
-	if(stored_research)
-		to_chat(H, span_notice("Copying files..."))
-		if(do_after(H, S.s_delay, src) && G.candrain && src)
-			stored_research.copy_research_to(S.stored_research)
-	to_chat(H, span_notice("Data analyzed. Process finished."))
+	if(!do_after(H, 30 SECONDS, target = src))
+		return
+
+	SSresearch.science_tech.modify_points_all(0)
```

Comment:
Can you make this instead take modify the techweb the server in question is connected to

Reply:
> Can you make this instead take modify the techweb the server in question is connected to

yeah

Reply:
just replace ``SSreseach.science_tech`` with ``stored_research``